### PR TITLE
theharvester: 3.2.3 -> 4.0.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 4.0.0

Change log: https://github.com/laramies/theHarvester/releases/tag/4.0.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129474` at a36ab913 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/67bd0fb8eba437bf807b0a749b818352">python38Packages.dependency-injector</a></li>
    <li><a href="https://gist.github.com/52fdfce9a958d3d6a94b4a2feb15996c">python39Packages.dependency-injector</a></li>
  </ul>
</details>
<details>
  <summary>12 packages built successfully:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.graphene</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.slowapi</li>
    <li>python38Packages.starlette</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.graphene</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.slowapi</li>
    <li>python39Packages.starlette</li>
    <li>theharvester</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/security/theharvester/default.nix [hunk 1]
```diff
@@ -33,14 +33,23 @@ python3.pkgs.buildPythonApplication rec {
...
+   pytest
+   pytest-asyncio
+  ];
 
-  checkPhase = "runHook preCheck ; pytest tests/test_myparser.py ; runHook postCheck";
   # We don't run other tests (discovery modules) because they require network access
+  checkPhase = ''
+  runHook preCheck
+  pytest tests/test_myparser.py
+  runHook postCheck
```
Comment (SuperSandro2000) [inline pkgs/tools/security/theharvester/default.nix:51]:
```suggestion
    runHook preCheck
    pytest tests/test_myparser.py
    runHook postCheck
```

File: pkgs/tools/security/theharvester/default.nix [hunk 2]
```diff
@@ -33,14 +33,23 @@ python3.pkgs.buildPythonApplication rec {
...
+    slowapi
+    starlette
+    uvicorn
     uvloop
   ];
 
-  checkInputs = [ python3.pkgs.pytest ];
+  checkInputs = with  python3.pkgs; [
+   pytest
+   pytest-asyncio
```
Comment (SuperSandro2000) [inline pkgs/tools/security/theharvester/default.nix:44]:
```suggestion
  checkInputs = with python3.pkgs; [
     pytest
     pytest-asyncio
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 129474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>4 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>esphome: log was empty</li>
<li>platformio: log was empty</li>
<li>python38Packages.dependency-injector: log was empty</li>
<li>python39Packages.dependency-injector: log was empty</li>
  </ul>
</details>
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>dyndnsc</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.graphene</li>
<li>python38Packages.json-logging</li>
<li>python38Packages.slowapi</li>
<li>python38Packages.starlette</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.graphene</li>
<li>python39Packages.json-logging</li>
<li>python39Packages.slowapi</li>
<li>python39Packages.starlette</li>
<li>theharvester</li>
</ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.slowapi</li>
    <li>python39Packages.slowapi</li>
    <li>theharvester</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 129474` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python38Packages.slowapi</li>
<li>python39Packages.slowapi</li>
<li>theharvester</li>
</ul>
</details>

**Pull request merged**